pkg/crypto/webkms/remotecrypto.go

```diff
@@ -0,0 +1,527 @@
+	if err != nil {
+		return nil, fmt.Errorf("posting ComputeMAC request failed [%s, %w]", destination, err)
+	}
+
+	// handle response
+	defer closeResponseBody(resp.Body, logger, "ComputeMAC")
+
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("read response for ComputeMAC failed [%s, %w]", destination, err)
+	}
+
+	httpResp := &computeMACResp{}
+
+	err = r.unmarshalFunc(respBody, httpResp)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal ComputeMAC response failed [%s, %w]", destination, err)
+	}
+
+	macBytes, err := base64.URLEncoding.DecodeString(httpResp.MAC)
+	if err != nil {
+		return nil, err
+	}
+
+	return macBytes, nil
+}
+
+// VerifyMAC remotely determines if mac is a correct authentication code (MAC) for data using a key at KeyURL
+// using a matching MAC primitive in kh key handle and returns nil if so, otherwise it returns an error.
+func (r *RemoteCrypto) VerifyMAC(mac, data []byte, keyURL interface{}) error {
+	destination := fmt.Sprintf("%s", keyURL) + verifyMACURI
+
+	vReq := verifyMACReq{
+		MAC:  base64.URLEncoding.EncodeToString(mac),
+		Data: base64.URLEncoding.EncodeToString(data),
+	}
+
+	httpReqBytes, err := r.marshalFunc(vReq)
+	if err != nil {
+		return fmt.Errorf("marshal request for VerifyMAC failed [%s, %w]", destination, err)
+	}
+
+	resp, err := r.postHTTPRequest(destination, httpReqBytes)
+	if err != nil {
+		return fmt.Errorf("posting VerifyMAC request failed [%s, %w]", destination, err)
+	}
+
+	// handle response
+	defer closeResponseBody(resp.Body, logger, "VerifyMAC")
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("posting VerifyMAC request returned http error: %s", resp.Status)
+	}
+
+	return nil
+}
+
+// WrapKey will remotely execute key wrapping of cek using apu, apv and recipient public key 'recPubKey'.
+// 'opts' allows setting the option sender key handle using WithSender() option where the sender key handle consists
+// of a remote key located in the option as a keyURL. This option allows ECDH-1PU key wrapping (aka Authcrypt).
+// The absence of this option uses ECDH-ES key wrapping (aka Anoncrypt).
+// 		RecipientWrappedKey containing the wrapped cek value
+// 		error in case of errors
+func (r *RemoteCrypto) WrapKey(cek, apu, apv []byte, recPubKey *crypto.PublicKey,
+	opts ...crypto.WrapKeyOpts) (*crypto.RecipientWrappedKey, error) {
+	destination := r.keystoreURL + wrapURI
+
+	pOpts := crypto.NewOpt()
+
+	for _, opt := range opts {
+		opt(pOpts)
+	}
+
+	senderURL := pOpts.SenderKey()
+	recipientPubKey := PubKeyToSerializableReq(recPubKey)
+	wReq := &WrapKeyReq{
+		CEK:       base64.URLEncoding.EncodeToString(cek),
+		APU:       base64.URLEncoding.EncodeToString(apu),
+		APV:       base64.URLEncoding.EncodeToString(apv),
+		RecPubKey: recipientPubKey,
+	}
+
+	// if senderURL is set, extract keyID and add it to the request (for ECDH-1PU wrapping)
+	if senderURL != "" {
+		senderURLStr := fmt.Sprintf("%s", senderURL)
```

Comment:
nit: no need for `fmt.Sprintf`

Reply:
senderURL type is `interface{}`, it needs to be translated to string.